src/JsonReader.cs

```diff
@@ -542,127 +564,169 @@ sealed class NonProperty : IJsonProperty<Unit, JsonReadResult<Unit>>
+                                || TrySetPropertyIndex(11, property11, ref reader, ref currentPropertyIndex)
+                                || TrySetPropertyIndex(12, property12, ref reader, ref currentPropertyIndex)
+                                || TrySetPropertyIndex(13, property13, ref reader, ref currentPropertyIndex)
+                                || TrySetPropertyIndex(14, property14, ref reader, ref currentPropertyIndex)
+                                || TrySetPropertyIndex(15, property15, ref reader, ref currentPropertyIndex)
+                                || TrySetPropertyIndex(16, property16, ref reader, ref currentPropertyIndex);
+
+                            sm.OnPropertyNameRead();
 
-                    if (!reader.Read())
-                        throw PartialJsonNotSupportedException();
+                            break;
+                        }
 
-                    switch (property.Reader.TryRead(ref reader))
-                    {
-                        case { Incomplete: true }:
-                            throw PartialJsonNotSupportedException();
-                        case { Error: { } err }:
-                            error = err;
-                            return false;
-                        case { Value: var val }:
-                            value = (true, val);
-                            return true;
+                        case ObjectReadStateMachine.ReadResult.PropertyValue:
+                        {
+                            static JsonReadResult<TResult>? ReadPropertyValue<T>(ref Utf8JsonReader reader,
+                                                                                 IJsonProperty<T, JsonReadResult<T>> property,
+                                                                                 ref (bool, T) value,
+                                                                                 ref ObjectReadStateMachine sm,
+                                                                                 in ObjectReadState<T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16> state)
+                            {
+                                switch (property.Reader.TryRead(ref reader))
+                                {
+                                    case { Incomplete: true }:
+                                        return reader.Suspend((sm, state));
+                                    case { Error: { } err }:
+                                        return new JsonReadError(err);
+                                    case { Value: var val }:
+                                        value = (true, val);
+                                        return null;
+                                }
+                            }
+
+                            if (state.CurrentPropertyIndex is { } nextPropertyIndex)
+                            {
+                                var result = nextPropertyIndex switch
```

Comment:
nit: _current_ and _next_ seem confusing for the same thing so consider just saying `propertyIndex` here?

```suggestion
                            if (state.CurrentPropertyIndex is { } propertyIndex)
                            {
                                var result = propertyIndex switch
```